spring-cloud-gcp-storage/src/test/java/org/springframework/cloud/gcp/storage/GoogleStorageTests.java

```diff
@@ -50,6 +55,32 @@ public void testValidObject() throws Exception {
 		Assert.assertEquals(4096L, this.remoteResource.contentLength());
 	}
 
+	@Test
+	public void testWritable() throws Exception {
+		Assert.assertTrue(this.remoteResource instanceof WritableResource);
+		WritableResource writableResource = (WritableResource) this.remoteResource;
+		Assert.assertTrue(writableResource.isWritable());
+		OutputStream os = writableResource.getOutputStream();
+		Assert.assertTrue(os instanceof GoogleStorageOutputStream);
+	}
+
+	@Test
+	public void testGcsOutputStream() throws Exception {
+		Blob blob = Mockito.mock(Blob.class);
+		WriteChannel writeChannel = Mockito.mock(WriteChannel.class);
+		Mockito.when(blob.writer()).thenReturn(writeChannel);
+
+		GoogleStorageOutputStream os = new GoogleStorageOutputStream(blob);
+
+		// test single byte
+		os.write(99);
+		Mockito.verify(writeChannel).write(ByteBuffer.wrap(new byte[] { 99 }));
```

Comment:
Would be interesting to read and then write. I guess that makes more sense on an integration test, though.